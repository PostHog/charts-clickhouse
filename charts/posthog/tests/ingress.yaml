suite: PostHog ingress definition
templates:
  - templates/ingress.yaml

tests:
  - it: should be empty if ingress.enabled is set to false
    set:
      ingress.enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should have the correct apiVersion
    asserts:
      - hasDocuments:
          count: 1
      - isAPIVersion:
          of: networking.k8s.io/v1

  - it: should be the correct kind
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Ingress

  - it: the "spec" path should match the snapshot when using default values
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot:
          # Unfortunately we can't match the whole manifest as there are few
          # metadata values linked to the chart version
          path: spec

  - it: AWS - 'metadata.annotations' should match the default values
    set:
      cloud: "aws"
      ingress:
        hostname: <your-hostname>
        nginx:
          enabled: true
      cert-manager:
        enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            kubernetes.io/ingress.class: nginx
            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
            nginx.ingress.kubernetes.io/ssl-redirect: "true"
            cert-manager.io/cluster-issuer: "letsencrypt-prod"

  - it: DO - 'metadata.annotations' should match the default values
    set:
      cloud: "do"
      ingress:
        hostname: <your-hostname>
        nginx:
          enabled: true
      cert-manager:
        enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            kubernetes.io/ingress.class: nginx
            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
            nginx.ingress.kubernetes.io/ssl-redirect: "true"
            cert-manager.io/cluster-issuer: "letsencrypt-prod"

  - it: GCP - 'metadata.annotations' should match the default values
    set:
      cloud: "gcp"
      ingress:
        hostname: <your-hostname>
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            kubernetes.io/ingress.class: gce
            kubernetes.io/ingress.global-static-ip-name: posthog
            networking.gke.io/managed-certificates: RELEASE-NAME-gke-cert
            networking.gke.io/v1beta1.FrontendConfig: RELEASE-NAME-frontend-config

  - it: GCP - 'metadata.annotations' the ingress.gcp.ip_name override should work
    set:
      cloud: "gcp"
      ingress:
        hostname: <your-hostname>
        gcp:
          ip_name: ip_custom_name
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            kubernetes.io/ingress.class: gce
            kubernetes.io/ingress.global-static-ip-name: ip_custom_name
            networking.gke.io/managed-certificates: RELEASE-NAME-gke-cert
            networking.gke.io/v1beta1.FrontendConfig: RELEASE-NAME-frontend-config

  - it: NGINX - 'metadata.annotations' should match the default values
    set:
      ingress.nginx.enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            kubernetes.io/ingress.class: nginx
            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
            nginx.ingress.kubernetes.io/ssl-redirect: "true"

  - it: NGINX with Let's Encrypt - 'metadata.annotations' should match the default values
    set:
      ingress.nginx.enabled: true
      ingress.letsencrypt: true
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            kubernetes.io/ingress.class: nginx
            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
            nginx.ingress.kubernetes.io/ssl-redirect: "true"
            cert-manager.io/cluster-issuer: "letsencrypt-prod"

  - it: overrides to 'metadata.annotations' via 'ingress.annotations' should work
    set:
      ingress.annotations:
        key1: value1
        key2: value2
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            key1: value1
            key2: value2

  - it: AWS - 'spec.ingressClassName' and 'metadata.annotations.kubernetes.io/ingress.class' should be set
    set:
      cloud: "aws"
      ingress:
        hostname: <your-hostname>
        nginx:
          enabled: true
      cert-manager:
        enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            cert-manager.io/cluster-issuer: letsencrypt-prod
            kubernetes.io/ingress.class: nginx
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
            nginx.ingress.kubernetes.io/ssl-redirect: "true"
      - equal:
          path: spec.ingressClassName
          value: nginx

  - it: GCP - custom 'spec.ingressClassName' and 'metadata.annotations.kubernetes.io/ingress.class' should be set
    set:
      cloud: "gcp"
      ingress:
        hostname: <your-hostname>
        nginx:
          enabled: false
        class: test
      cert-manager:
        enabled: false
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations
          value:
            kubernetes.io/ingress.class: test
            kubernetes.io/ingress.global-static-ip-name: posthog
            meta.helm.sh/release-name: RELEASE-NAME
            meta.helm.sh/release-namespace: NAMESPACE
            networking.gke.io/managed-certificates: RELEASE-NAME-gke-cert
            networking.gke.io/v1beta1.FrontendConfig: RELEASE-NAME-frontend-config
      - equal:
          path: spec.ingressClassName
          value: test
