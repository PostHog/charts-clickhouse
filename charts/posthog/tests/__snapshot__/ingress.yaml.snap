the "spec" path should match the snapshot when splitting decide traffic out:
  1: |
    rules:
    - http:
        paths:
        - backend:
            service:
              name: RELEASE-NAME-posthog-web
              port:
                number: 8000
          path: /
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /batch
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /capture/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /capture
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-decide
              port:
                number: 8000
          path: /decide/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-decide
              port:
                number: 8000
          path: /decide
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /e/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /e
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /engage/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /engage
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /track/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /track
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /s/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /s
          pathType: Exact
the "spec" path should match the snapshot when splitting recordings traffic out:
  1: |
    rules:
    - http:
        paths:
        - backend:
            service:
              name: RELEASE-NAME-posthog-web
              port:
                number: 8000
          path: /
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /batch
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /capture/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /capture
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /decide/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /decide
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /e/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /e
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /engage/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /engage
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /track/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /track
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-recordings
              port:
                number: 8000
          path: /s/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-recordings
              port:
                number: 8000
          path: /s
          pathType: Exact
the "spec" path should match the snapshot when using default values:
  1: |
    rules:
    - http:
        paths:
        - backend:
            service:
              name: RELEASE-NAME-posthog-web
              port:
                number: 8000
          path: /
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /batch
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /capture/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /capture
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /decide/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /decide
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /e/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /e
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /engage/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /engage
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /track/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /track
          pathType: Exact
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /s/
          pathType: Prefix
        - backend:
            service:
              name: RELEASE-NAME-posthog-events
              port:
                number: 8000
          path: /s
          pathType: Exact
